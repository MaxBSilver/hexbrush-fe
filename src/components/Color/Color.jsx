import React, { Component } from 'react';

class Color extends Component {
	state = {
		hex: '#FFFFFF',
		locked: false
	};

	render() {
		const { locked, hex } = this.state;
		return (
			<section className="Color">
				<p className="Color-hex">{hex}</p>
				<i className="Color-lock-icon material-icons" onClick={this.setState({ locked: !locked })}>
					{locked ? 'lock' : 'lock_open'}
				</i>
			</section>
		);
	}
}

export default Color;
