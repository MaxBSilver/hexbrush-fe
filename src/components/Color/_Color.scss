.Color {
	display: flex;
	justify-content: flex-end;
	align-items: center;
	flex-direction: column;
	padding: 25px;
	color: #b6b6b69a;

	&:first-child {
		border-radius: 8px 0 0 8px;
	}

	&:nth-child(5) {
		border-radius: 0 8px 8px 0;
	}
}

.Color-hex {
	cursor: default;
}

.Color-lock-icon {
	font-size: 50px;
	cursor: pointer;
  transition: background-color 0.1s;
  padding: 3px;
  border-radius: 5px;

	&:hover {
    background-color: #b6b6b66b;
	}
}
