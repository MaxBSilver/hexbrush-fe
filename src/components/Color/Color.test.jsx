import React from 'react';
import Color from './Color';
import { shallow } from 'enzyme';

describe('Color', () => {
	let wrapper, mockLockColor;
	beforeEach(() => {
		mockLockColor = jest.fn();
		wrapper = shallow(<Color isLocked={false} hex="#fa5b27" id={1} lockColor={mockLockColor} />);
	});

	it('should match the snapshot if unlocked', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should match the snapshot if locked', () => {
		const wrapper = shallow(<Color isLocked={true} hex="#fa5b27" id={1} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should call props.lockColor when the icon is clicked', () => {
		wrapper.find('.Color-lock-icon').simulate('click');
		expect(mockLockColor).toHaveBeenCalledWith(1);
	});
});
