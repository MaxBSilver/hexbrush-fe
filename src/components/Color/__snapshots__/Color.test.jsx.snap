// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color should match the snapshot if locked 1`] = `ShallowWrapper {}`;

exports[`Color should match the snapshot if unlocked 1`] = `ShallowWrapper {}`;
